### ISSUE TITLE
Enhancement, JetInstanceAware

### ISSUE THREAD

**Issue by neilstevenson** [hazelcast/hazelcast-jet#2549]
IMDG provides https://docs.hazelcast.org/docs/4.0.3/javadoc/com/hazelcast/core/HazelcastInstanceAware.html

A class such as Runnable that implements `HazelcastInstanceAware` has this method

```
    @Override
    public void setHazelcastInstance(HazelcastInstance arg0) {
        this.hazelcastInstance = arg0;
    }
```

However, this is no equivalent for Jet

Being able to do so would enable a Runnable/Callable to be able to elegantly check which Jobs are running, etc

**Comment by viliam-durina** [comment]
It can only work for Runnables submitted to Jet. IMDG isn't aware of the enclosing `JetInstance`.

**Comment by neilstevenson** [comment]
@viliam-durina Good point. I've logged this against IMDG as a linked issue. Here I was asking for `JetInstanceAware` for a Runnable/Callable sent to the executor service, but more generally it would be useful to have the reverse for `JetInstance.getHazelcastInstance()`, which would make this issue obsolete.

**Comment by viliam-durina** [comment]
@neilstevenson It would be possible, but since IMDG doesn't depend on jet, the method signature would be:
```java
public interface HazelcastInstance {
    /**
     * Returns the JetInstance, if this cluster is a Jet cluster. You can safely cast it to `com.hazelcast.jet.JetInstance`.
     */
    @Nullable
    Object getJetInstance();
}
```
